include/Core/Base/Mixin/Renderer.php

```diff
@@ -345,6 +345,13 @@ protected function render_filtering_table_header() {
 		bd_render_filtering_table_header();
 	}
 
+	/**
+	 * Render attachment settings.
+	 */
+	protected function render_attachment_settings() {
+		do_action( 'bd_render_attachment_settings', $this->field_slug, $this->item_type );
```

Comment:
Pass the entire module object as a parameter. (ie) Just pass `$this` as a parameter to the action.

Also add PHPDoc block to the action.